crates/iceberg/src/spec/values.rs

```diff
@@ -0,0 +1,663 @@
+                PrimitiveType::Float => Ok(Value::Float(OrderedFloat(f32::from_le_bytes(
+                    bytes.try_into()?,
+                )))),
+                PrimitiveType::Double => Ok(Value::Double(OrderedFloat(f64::from_le_bytes(
+                    bytes.try_into()?,
+                )))),
+                PrimitiveType::Date => Ok(Value::Date(date::days_to_date(i32::from_le_bytes(
+                    bytes.try_into()?,
+                ))?)),
+                PrimitiveType::Time => Ok(Value::Time(time::microseconds_to_time(
+                    i64::from_le_bytes(bytes.try_into()?),
+                )?)),
+                PrimitiveType::Timestamp => Ok(Value::Timestamp(
+                    timestamp::microseconds_to_datetime(i64::from_le_bytes(bytes.try_into()?))?,
+                )),
+                PrimitiveType::Timestamptz => Ok(Value::TimestampTZ(
+                    timestamptz::microseconds_to_datetimetz(i64::from_le_bytes(bytes.try_into()?))?,
+                )),
+                PrimitiveType::String => Ok(Value::String(std::str::from_utf8(bytes)?.to_string())),
+                PrimitiveType::Uuid => Ok(Value::UUID(Uuid::from_u128(u128::from_be_bytes(
+                    bytes.try_into()?,
+                )))),
+                PrimitiveType::Fixed(len) => Ok(Value::Fixed(*len as usize, Vec::from(bytes))),
+                PrimitiveType::Binary => Ok(Value::Binary(Vec::from(bytes))),
+                _ => Err(Error::new(
+                    crate::ErrorKind::DataInvalid,
+                    "Converting bytes to decimal is not supported.",
+                )),
+            },
+            _ => Err(Error::new(
+                crate::ErrorKind::DataInvalid,
+                "Converting bytes to non-primitive types is not supported.",
+            )),
+        }
+    }
+
+    /// Get datatype of value
+    pub fn datatype(&self) -> Type {
+        match self {
+            Value::Boolean(_) => Type::Primitive(PrimitiveType::Boolean),
+            Value::Int(_) => Type::Primitive(PrimitiveType::Int),
+            Value::Long(_) => Type::Primitive(PrimitiveType::Long),
+            Value::Float(_) => Type::Primitive(PrimitiveType::Float),
+            Value::Double(_) => Type::Primitive(PrimitiveType::Double),
+            Value::Date(_) => Type::Primitive(PrimitiveType::Date),
+            Value::Time(_) => Type::Primitive(PrimitiveType::Time),
+            Value::Timestamp(_) => Type::Primitive(PrimitiveType::Timestamp),
+            Value::TimestampTZ(_) => Type::Primitive(PrimitiveType::Timestamptz),
+            Value::Fixed(len, _) => Type::Primitive(PrimitiveType::Fixed(*len as u64)),
+            Value::Binary(_) => Type::Primitive(PrimitiveType::Binary),
+            Value::String(_) => Type::Primitive(PrimitiveType::String),
+            Value::UUID(_) => Type::Primitive(PrimitiveType::Uuid),
+            Value::Decimal(dec) => Type::Primitive(PrimitiveType::Decimal {
+                precision: 38,
```

Comment:
I think we want to make the precision configurable as well.

Reply:
I think we then might need to store the precision in another field. I don't know how to calculate the precision from the scale and I don't see a way to store it in the rust decimal.